### PR TITLE
[Android] Pager progress indicator component

### PR DESCRIPTION
**Pull request by ab-gnm**
#### Description

**Figma**: https://www.figma.com/design/NMEYESZ2PNMvA1Tziz9RzL/Image-carousel?node-id=11-4&t=Z2tEbwfcPC0APnxb-0

**Key changes**:

1. Created an _internal_ component `PagerProgressItem` that draws a single item in the progress indicator. The component is responsible for size, shape, colour and animations of the individual item.
2. Created the `PagerProgressIndicator` component to represent a progress indicator with a row of `PagerProgressItem`s. This is a basic component that is available publicly. 
   * This is highly customisable including scale factors, colours, size, shape, etc.
   * This can be also used to replace current pager indicators as is (by setting `unselectedItemScaleFactor` to `1f`).
3. Created the `PagerProgressBar` to provide full functionality of the pager as per specs. 
   * The bar shows pager `PagerProgressIndicator` horizontally centered
   * On tablets, we also display buttons for going forward and backward in the pager
   * Only button and indicator colours are configurable. Size, shape, spacing are fixed according to Source figma specs
4. Updated `SourceBaseIconButton` so it doesn't have any dependency on `priority` or `theme`. This makes it completely independent icon that just applies the correct shape and size, and let's the developer provide all the colours and icon. The `SourceIconButton` now converts theme + priority into correct colours and calls `SourceBaseIconButton`
5. Added two new vector icons for next/prev buttons

#### Checklist
- [X] Changes have been checked by the developer
- [ ] Changes have been checked by the reviewers
- [X] Unit tested

##### Recommended reviewiers

- Design review for UI changes from @guardian/design-system
- Code review from @guardian/android-developers or @guardian/ios-developers
- Optional code/API review from @guardian/client-side-infra

#### For pull requests introducing UI changes:

- [ ] Sign-off by Design: @guardian/design-system 
- [X] Dark Mode 
- [X] Tablet 
- [ ] Accessibility (e.g. VoiceOver): 

##### Screenshots or videos:

| Phone | Tablet |
|--------|--------|
| phone-light | tablet-light |
| phone-dark | tablet-dark |
| <video src="https://github.com/user-attachments/assets/52ceca8d-0680-42b2-8857-9f528bb8de29">| <video src="https://github.com/user-attachments/assets/437e444e-ae8a-4837-922d-6bd6e9669456"> |

### CODE REVIEW

File: android/gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [versions]
 group = "com.gu.source"
-libraryVersion = "0.2.2"
+libraryVersion = "0.3.0"
```
Comment (ab-gnm) [inline android/gradle/libs.versions.toml:3]: Incremented version to indicate new component

File: android/sample/src/main/kotlin/com/gu/source/ButtonPreview.kt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import com.gu.source.components.buttons.SourceIconButton
+import com.gu.source.daynight.AppColour
+import com.gu.source.icons.Check
+import com.gu.source.presets.palette.Culture200
+import com.gu.source.presets.palette.Culture800
+import com.gu.source.presets.typography.TextSansBold17
+import com.gu.source.theme.ReaderRevenueTheme
+
+@Composable
+internal fun ButtonPreview(modifier: Modifier = Modifier) {
```
Comment (ab-gnm) [inline android/sample/src/main/kotlin/com/gu/source/ButtonPreview.kt:22]: Extracted from `MainActivity.kt` as more components are added.

File: android/sample/src/main/kotlin/com/gu/source/ImagePagerWithProgressIndicator.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import com.gu.source.daynight.AppColour
+import com.gu.source.daynight.AppColourMode
+import com.gu.source.presets.palette.Neutral10
+import com.gu.source.presets.palette.Neutral100
+import com.gu.source.utils.PhoneBothModePreviews
+import com.gu.source.utils.TabletBothModePreviews
+
+@Suppress("MagicNumber")
+@Composable
+internal fun ImagePagerWithProgressIndicator(modifier: Modifier = Modifier) {
```
Comment (ab-gnm) [inline android/sample/src/main/kotlin/com/gu/source/ImagePagerWithProgressIndicator.kt:26]: Just a preview of the pager with some images in the sample app.

File: android/sample/src/main/kotlin/com/gu/source/MainActivity.kt [hunk 4]
```diff
@@ -36,32 +31,49 @@ class MainActivity : ComponentActivity() {
...
     val coroutineScope = rememberCoroutineScope()
     val scaffoldState = rememberBottomSheetScaffoldState()
 
+    var sheetContentType by remember {
+        mutableStateOf(SheetContentType.Palette)
+    }
+
     BottomSheetScaffold(
-        sheetContent = { Palette() },
+        sheetContent = {
```
Comment (ab-gnm) [inline android/sample/src/main/kotlin/com/gu/source/MainActivity.kt:57]: Slightly refactored the sample. Sheet content shows previews while the base scaffold just displays buttons to select different types of previews.
The sample needs more love as we add more components - that'd be done in another PR.

File: android/source/src/main/kotlin/com/gu/source/components/pager/PagerProgressIndicator.kt [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+@Preview
+@Composable
+internal fun PagerProgressIndicatorPreview() {
+    Column(
+        verticalArrangement = Arrangement.spacedBy(8.dp),
+        modifier = Modifier
+            .background(Color.White)
+            .padding(8.dp),
+    ) {
+        Column(modifier = Modifier.padding(8.dp)) {
```
Comment (michaelclapham) [inline android/source/src/main/kotlin/com/gu/source/components/pager/PagerProgressIndicator.kt:171]: Could we use a `repeat { index ->` here to avoid the need to copy paste this 5 times?

**Review by michaelclapham — APPROVED** [review]
Overall this looks great! I left only one tiny nit comment, but completely not a blocker.

We are just about to build something on Feast that requires this exact component! @ab-gnm could we have a chat at some point around how we could go about integrating this library into feast so we can also benefit from these components?

**Comment by ab-gnm** [comment]
@guardian/design-system I'm going to merge this PR and release the changes so we can start updating the image carousels in the app. This is still open for design feedback though. I'll include any design feedback in the next point release.
